content/master/concepts/patch-and-transform.md

```diff
@@ -0,0 +1,1692 @@
+associated composite resource or managed resources. 
+
+Examples of using patch and transforms include:
+ * changing the name of the external resource
+ * mapping generic terms like "east" or "west" to specific provider locations
+ * appending custom labels or strings to resource fields
+
+
+{{<hint "note" >}}
+<!-- vale alex.Condescending = NO -->
+Crossplane expects patch and transform operations to be simple changes.  
+Use [Composition Functions]({{<ref "./composition-functions">}}) for more
+complex or programmatic modifications.
+<!-- vale  alex.Condescending = YES -->
+{{</hint >}}
+
+
+A Composition [patch](#create-a-patch) is the action of changing a field.  
+A Composition [transform](#transform-a-patch) modifies the values before 
+applying the patch.
+
+## Create a patch
+
+Patches are part of an individual 
+{{<hover label="createComp" line="4">}}resource{{</hover>}} inside a 
+{{<hover label="createComp" line="2">}}Composition{{</hover>}}.
+
+The {{<hover label="createComp" line="8">}}patches{{</hover>}} field takes a
+list of patches to apply to the individual resource. 
+
+Each patch has a {{<hover label="createComp" line="9">}}type{{</hover>}}, which
+defines what kind of patch action Crossplane applies. 
+
+Patches reference fields inside a composite resource or Composition differently
+depending on the patch type, but all patches reference a 
+{{<hover label="createComp" line="10">}}fromFieldPath{{</hover>}} and
+{{<hover label="createComp" line="11">}}toFieldPath{{</hover>}}.
+
+The {{<hover label="createComp" line="10">}}fromFieldPath{{</hover>}} defines
+the patch's input values. 
+The {{<hover label="createComp" line="11">}}toFieldPath{{</hover>}} defines the
+data to change with a patch.
+
+Here is an example patch applied to a resource in a Composition. 
+```yaml {label="createComp",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: my-composed-resource
+      base:
+        # Removed for brevity
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.field1
+          toFieldPath: metadata.labels["patchLabel"]
+```
+
+### Selecting fields
+
+Crossplane selects fields in a composite resource or managed
+resource with
+a subset of 
+[JSONPath selectors](https://kubernetes.io/docs/reference/kubectl/jsonpath/),
+called "field selectors."
+
+Field selectors can select any field in a composite resource or managed resource 
+object, including the `metadata`, `spec` or `status` fields. 
+
+Field selectors can be a string matching a field name or an array index, in
+brackets. Field names may use a `.` character to select child elements.
+
+
+{{< expand "Example field selectors" >}}
+Here are some example selectors from a composite resource object.
+{{<table "table table-hover" >}}
+| Selector | Selected element | 
+| --- | --- |
+| `kind` | {{<hover label="select" line="3">}}kind{{</hover>}} |
```

Comment:
(1) This is going to require some gross hacking since the table and table highlighting is generated by our CSS library (bootstrap) while the code highlight is a hugo injected thing. I don't think we could do this without making some custom table codes or something. For now I turned off the row highlighting to make it more likely the user will hover the keyword.

(2) ✅